src/registries/npm-registry.js

```diff
@@ -208,26 +208,33 @@ export default class NpmRegistry extends Registry {
       return this.token;
     }
 
-    const registry = this.getRegistry(packageName);
+    const baseRegistry = this.getRegistry(packageName);
+    const registries = [baseRegistry];
 
-    // Check for bearer token.
-    const authToken = this.getRegistryOrGlobalOption(registry, '_authToken');
-    if (authToken) {
-      return `Bearer ${String(authToken)}`;
+    if (baseRegistry === `https://registry.yarnpkg.com/`) {
```

Comment:
So if package comes from `registry.yarnpkg.com` then you want to add `registry.npmjs.org` to the list of registries in the attempt of getting username and password.

That looks like a lazy patch, it would be better to set up `getRegistryOrGlobalOption` to return same data for yarnpkg and npmjs rather than patch every callsite